CONTRIBUTING.md

```diff
@@ -1,29 +1,123 @@
+employer) retain the copyright to your contribution; the CLA simply gives us
+permission to use and redistribute your contributions as part of the project.
+Please visit https://cla.developers.google.com/ to see your current agreements
+on file or to sign a new one. You generally only need to submit a Google CLA
+once, so if you've already submitted one (even if it was for a different
+project), you probably don't need to do it again.
 
-## Code reviews
+> [!WARNING]
+> Please note carefully clauses [#5](https://cla.developers.google.com/about/google-corporate#:~:text=You%20represent%20that%20each%20of%20Your%20Contributions%20is%20Your%20original%20creation)
+> and [#7](https://cla.developers.google.com/about/google-corporate#:~:text=Should%20You%20wish%20to%20submit%20work%20that%20is%20not%20Your%20original%20creation%2C%20You%20may%20submit%20it%20to%20Google%20separately)
+> in the CLA. Any code that you contribute to this project must be **your**
+> original creation. Code generated by artificial intelligence tools **does
+> not** qualify as your original creation.
+
+### Review our community guidelines
+
+We have a [code of conduct](CODE_OF_CONDUCT.md) to make the ReCirq project an open
+and welcoming community environment. Please make sure to read and abide by the
+code of conduct.
+
+## Contribution process
 
 All submissions, including submissions by project members, require review. We
-use GitHub pull requests for this purpose. Consult
-[GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
-information on using pull requests.
+use the tools provided by GitHub for [pull
+requests](https://help.github.com/articles/about-pull-requests/) for this
+purpose. The preferred manner for submitting pull requests is to fork the ReCirq
+[repository](https://github.com/quantumlib/ReCirq), create a [git
+branch](https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell) in
+this fork to do your work, and when ready, create a pull request from this
+branch to the main ReCirq repository. The subsections below describe the process
+in more detail.
+
+Pleae make sure to follow the [Google Style
+Guides](https://google.github.io/styleguide/) in your code, particularly the
+[style guide for Python](https://google.github.io/styleguide/pyguide.html).
+
+### Repository forks
+
+1.  Fork the ReCirq repository (you can use the _Fork_ button in upper right
+    corner of the [repository page](https://github.com/quantumlib/ReCirq)).
+    Forking creates a new GitHub repo at the location
+    `https://github.com/USERNAME/ReCirq`, where `USERNAME` is your GitHub user
+    name.
+
+1.  Clone (using `git clone`) or otherwise download your forked repository to
+    your local computer, so that you have a local copy where you can do your
+    development work using your preferred editor and development tools.
+
+1.  Check out the `main` branch and create a new git branch from `main`:
```

Comment:
The mainline branch in this repository is `master`, please update here and below.

Alternatively, you can switch the default branch to the recommended `main`, but that should be done in a separate PR.

Cirq had this switched earlier in https://github.com/quantumlib/Cirq/pull/6363.

Reply:
The instructions still refer to the `main` branch which does not exist.

Can you please comment if you plan on renaming `master` to `main` in this repo soon?
If so the text can stay as is.  Otherwise we should use `master` now and replace it with `main` later after the rename.

Reply:
Ack, foo. Switching between projects has fried my ability to remember which ones use which. (And yes, I think eventually we should rename master to main, but not this week, so I'll fix this up instead.)